xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-api/src/main/java/org/xwiki/mentions/notifications/MentionNotificationParameters.java

```diff
@@ -19,105 +19,161 @@
-        this.anchorId = anchorId;
-        this.xdom = xdom;
+        this.version = version;
     }
 
     /**
-     *
      * @return the reference of the author of the mention
      */
-    public DocumentReference getAuthorReference()
+    public String getAuthorReference()
     {
         return this.authorReference;
     }
 
     /**
-     *
      * @return the document in which the mention has been done
      */
-    public DocumentReference getDocumentReference()
+    public EntityReference getEntityReference()
+    {
+        return this.entityReference;
+    }
+
+    /**
+     * @return the location of the mention
+     */
+    public MentionLocation getLocation()
     {
-        return this.documentReference;
+        return this.location;
     }
 
     /**
+     * Add a to the map of mentions.
      *
-     * @return the identity of the mentioned user
+     * @param type the type of the mentioned user
+     * @param mentionedActorReference the mention notification parameters
+     * @return the current object
      */
-    public DocumentReference getMentionedIdentity()
+    public MentionNotificationParameters addMention(String type,
+        MentionNotificationParameter mentionedActorReference)
     {
-        return this.mentionedIdentity;
+        addToMap(type, mentionedActorReference, this.mentions);
+        return this;
     }
 
     /**
+     * Add a mention to the map of new mentions.
      *
-     * @return the location of the mention
+     * @param type the type of the mentioned user
+     * @param mentionedActorReference the mention notification parameter
+     * @return the current object
      */
-    public MentionLocation getLocation()
+    public MentionNotificationParameters addNewMention(String type,
+        MentionNotificationParameter mentionedActorReference)
     {
-        return this.location;
+        addToMap(type, mentionedActorReference, this.newMentions);
+        return this;
     }
 
     /**
+     * Returns a map of the new mentions. The type of the mentioned actors are used as keys, and the values are
+     * {@link MentionNotificationParameter}, identifying a unique mention in a page by its actor reference and its
+     * anchor.
      *
-     * @return the anchor link to use
+     * @return the map of new mentions
      */
-    public String getAnchorId()
+    public Map<String, Set<MentionNotificationParameter>> getNewMentions()
     {
-        return this.anchorId;
+        return this.newMentions;
     }
 
     /**
+     * Returns a map of all the mentions, including the new ones. The type of the mentioned actors are used as keys, and
+     * the values are {@link MentionNotificationParameter}, identifying a unique mention in a page by its actor
+     * reference and its anchor.
      *
-     * @return the content xdom
+     * @return the map of all the mentions
+     */
+    public Map<String, Set<MentionNotificationParameter>> getMentions()
+    {
+        return this.mentions;
```

Comment:
Wondering if it wouldn't be safer here to return an immutable map of the mentions: those are not supposed to be modified externally.

Reply:
Interesting point. I'll add a `lock()` operation. When `lock` is called, `addMention` and `addNewMention` cannot add new `MentionNotificationParameter` objects to the maps. In addition I'll make `getMentions` and `getNewMentions` return unmodifiable maps.

Reply:
> I'll add a lock() operation. When lock is called, addMention and addNewMention cannot add new MentionNotificationParameter objects to the maps

Not sure that's needed but why not. 

> In addition I'll make getMentions and getNewMentions return unmodifiable maps.

ok, I suggested this one because it's a common mistake to call update operation on a map returned by an API.

Reply:
ok, I've limited my changes to making the returned maps unmodifiable.